### PR TITLE
Link and compile against rosbag2_storage_mcap: Fixed issue 1492

### PR DESCRIPTION
**Pull request by ahcorde**
Fixed issue https://github.com/ros2/rosbag2/issues/1492

How to test it 
```sh
rm -rf /tmp/colcon-example && mkdir /tmp/colcon-example && cd /tmp/colcon-example
mkdir -p src/example

cat > src/example/CMakeLists.txt <<'EOF'
cmake_minimum_required(VERSION 3.14)
project(example)

find_package(ament_cmake REQUIRED)
find_package(rosbag2_storage_mcap REQUIRED)

file(WRITE empty.cc "")
add_library(${PROJECT_NAME} SHARED empty.cc)

target_link_libraries(${PROJECT_NAME} rosbag2_storage_mcap::rosbag2_storage_mcap)

ament_package()
EOF

cat > src/example/package.xml <<'EOF'
<?xml version="1.0"?>
<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
  <name>example</name>
  <version>0.0.1</version>
  <description>thing</description>
  <maintainer email="meh@nowhere.com">Meh</maintainer>
  <license>Apache-2.0</license>
  <buildtool_depend>ament_cmake</buildtool_depend>
  <depend>rosbag2_storage_map</depend>
  <export>
    <build_type>ament_cmake</build_type>
  </export>
</package>
EOF

source /opt/ros/humble/setup.bash
colcon build
```

FYI @EricCousineau-TRI

### CODE REVIEW

**Comment by MichaelOrlov** [comment]
@ahcorde Could you please retarget your PR to the Rolling branch?  
Our workflow for the fixes is the first fix on Rolling then backport to the other releases.

**Comment by ahcorde** [comment]
Sure @MichaelOrlov, Done

**Comment by ahcorde** [comment]
@MichaelOrlov some [tests](https://build.ros2.org/job/Rpr__rosbag2__ubuntu_jammy_amd64/1148/) are failing is this expected?

**Comment by MichaelOrlov** [comment]
> @MichaelOrlov some [tests](https://build.ros2.org/job/Rpr__rosbag2__ubuntu_jammy_amd64/1148/) are failing is this expected?

Actually not. Those types of test failures are not expected.

**Comment by MichaelOrlov** [comment]
@ros-pull-request-builder retest this please

**Comment by ahcorde** [comment]
CI is green, thank you @MichaelOrlov for launching CI again

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19794/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14297/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20518/)

**Comment by MichaelOrlov** [comment]
Re-run Windows CI
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20528/)

**Review by MichaelOrlov — COMMENTED** [review]
@ahcorde What was a rationale for installing and exporting content of the `include` folder?
There only one file with visibility macros.
https://github.com/ros2/rosbag2/tree/rolling/rosbag2_storage_mcap/include/rosbag2_storage_mcap

We didn't do this originally for a reason. i.e. it doesn't make any sense.

I would appreciate if you will remove it from delta.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19806/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14310/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20531/)

**Review by clalancette — APPROVED** [review]
Looks good to me, but let's wait for @MichaelOrlov to approve before merging.

**Review by MichaelOrlov — APPROVED** [review]
LGTM.

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport iron

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport humble

**Pull request merged**
